controllers/quay/quayregistry_controller.go

```diff
@@ -1055,30 +1079,38 @@ func (r *QuayRegistryReconciler) updateWithCondition(
 
 func (r *QuayRegistryReconciler) cleanupPreviousSecret(log logr.Logger, ctx context.Context, quay *v1.QuayRegistry) error {
 	log.Info("deleting old objects")
-	previousSecret := &corev1.Secret{}
-	if err := r.Client.Get(
-		ctx,
-		types.NamespacedName{
-			Name:      quay.Spec.ConfigBundleSecret,
-			Namespace: quay.GetNamespace(),
-		},
-		previousSecret,
-	); err != nil {
-		r.Log.Info(fmt.Sprintf("%s secret not found, skipping", quay.Spec.ConfigBundleSecret))
-	}
-	if err := r.Client.Delete(ctx, previousSecret); err != nil {
+
+	log.Info("getting current config bundle secret")
+	previousSecrets, err := r.GetOldConfigBundleSecret(ctx, quay)
+	if err != nil {
 		if !errors.IsNotFound(err) {
 			return r.updateWithCondition(
 				ctx,
 				quay,
 				v1.ConditionTypeRolloutBlocked,
 				metav1.ConditionTrue,
 				v1.ConditionReasonComponentCreationFailed,
-				fmt.Sprintf("could not delete old objects: %s", err),
+				fmt.Sprintf("could not get previous config bundle secret: %s", err),
+			)
+		}
+	}
+
+	log.Info("deleting previous config bundle secrets")
+	for _, secret := range previousSecrets {
+		if err := r.Client.Delete(ctx, &secret); err != nil {
+			log.Error(err, "failed to delete previous config bundle secret")
+			return r.updateWithCondition(
+				ctx,
+				quay,
+				v1.ConditionTypeRolloutBlocked,
+				metav1.ConditionTrue,
+				v1.ConditionReasonComponentCreationFailed,
+				fmt.Sprintf("could not delete previous config bundle secret: %s", err),
 			)
 		}
-		log.Info("previous secret not found, continuing")
+		log.Info("successfully deleted previous config bundle secret")
```

Comment:
this will print the same line for each secret, do we want to print the name of the secret to make this log more useful?

Reply:
cleaned up the logging here